File: 0014_Longest_Common_Prefix/solution.cs
/*
14. Longest Common Prefix

Write a function to find the longest common prefix string amongst an array of strings.

*/

public class Solution014 {
    public string LongestCommonPrefix(string[] strs) {
        if(strs == null || strs.Length == 0)  
        {  
            return "";  
        }  
        for(int i=0; i<strs[0].Length; i++)  
        {  
            for(int j=1; j<strs.Length; ++j)  
            {  
                if(strs[j].Length<=i||strs[j][i]!=strs[0][i])  
                {  
                    return strs[0].Substring(0,i);  
                }  
            }               
        }  
        return strs[0];  
    }
}

File: 0014_Longest_Common_Prefix/solution.java
package cn.huang.leetcode;

import java.util.Arrays;

/*
    14 Longest Common Prefix

    Write a function to find the longest common prefix string amongst an array of strings.

    If there is no common prefix, return an empty string "".

    Example 1:

    Input: ["flower","flow","flight"]
    Output: "fl"
    Example 2:

    Input: ["dog","racecar","car"]
    Output: ""
    Explanation: There is no common prefix among the input strings.
    Note:

    All given inputs are in lowercase letters a-z.
 */
public class LC_0014_LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0)
            return "";

        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < strs[0].length(); i++) {
            char c = strs[0].charAt(i);
            for (int j = 1; j < strs.length; j++) {
                if (i>= strs[j].length() || strs[j].charAt(i) != c)
                    return sb.toString();
            }
            sb.append(c);
        }
        return sb.toString();
    }

    /*
     * 对字符串数组进行排序，然后只要比较首尾两个字符串即可
     */
    public String longestCommonPrefix_v2(String[] strs) {
        StringBuilder result = new StringBuilder();

        if (strs != null && strs.length > 0) {

            Arrays.sort(strs);

            char[] a = strs[0].toCharArray();
            char[] b = strs[strs.length - 1].toCharArray();

            for (int i = 0; i < a.length; i++) {
                if (b.length > i && b[i] == a[i]) {
                    result.append(b[i]);
                } else {
                    return result.toString();
                }
            }
            return result.toString();
        }
        return "";
    }
}